src/main/scala/RockPaperScissors.scala

```diff
@@ -0,0 +1,76 @@
+import scala.io.StdIn.readLine
+
+object RockPaperScissors extends App {
+  println("please enter the amount of games you wish to play")
+  var totalGames = takeInput().toInt
+  var scissorsWeight = 0
+  var rockWeight = 0
+  var paperWeight = 0
+  println("please enter the mode ou wish to enter" )
+
+  //player vs computer
+  /*
+  while (totalGames > 0)
+  {
+
+    var moveAI = commitAImove(generateAIMove(), scissorsWeight, rockWeight, paperWeight)
+    var movePlayer = takeInput()
+    weightAdjustmentCheck(movePlayer)
+    println(determineWinner(moveAI, movePlayer ))
+    println("genrated Ai move " + moveAI)
+    println(rockWeight)
+    totalGames -= 1
+  }
+  */
+
+  //computer vs computer
+  while (totalGames > 0)
+  {
+
+    var moveAI = commitAImove(generateAIMove(), scissorsWeight, rockWeight, paperWeight)
+    var movePlayer = commitAImove(generateAIMove(), scissorsWeight, rockWeight, paperWeight)
+    weightAdjustmentCheck(movePlayer)
+    println(determineWinner(moveAI, movePlayer ))
+    println("genrated Ai move " + moveAI)
+    println(rockWeight)
+    totalGames -= 1
+  }
+
+  def takeInput(): String = readLine()
+
+  def generateAIMove(): Int =
```

Comment:
this again could be a one line method, not entirely sure why you're doing it in this way